e2e/rbd.go

```diff
@@ -263,33 +263,90 @@ var _ = Describe("RBD", func() {
 						e2elog.Logf("backend image count %d expected image count %d", len(images), 1)
 						Fail("validate backend image failed")
 					}
+					totalCount := 5
 					snap := getSnapshot(snapshotPath)
 					snap.Namespace = f.UniqueName
 					snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
-					err = createSnapshot(&snap, deployTimeout)
+					// create snapshot
+					for i := 0; i < totalCount; i++ {
+						snap.Name = fmt.Sprintf("%s%d", f.UniqueName, i)
+						err = createSnapshot(&snap, deployTimeout)
+						if err != nil {
+							Fail(err.Error())
+						}
+					}
+
+					imageList := listRBDImages(f)
+					// total images in cluster is 1 parent rbd image+ total snaps
+					if len(imageList) != totalCount+1 {
+						e2elog.Logf("backend images not matching kubernetes pvc,snap count,image count %d kubernetes resource count %d", len(imageList), totalCount+1)
```

Comment:
this error message has to be revisited or can be better formatted.

Reply:
Sure, can do a enhancement